catalog/nsili/catalog-nsili-sourcestoquery-ui/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -0,0 +1,41 @@
+<!--
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+-->
+<blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="
+       http://www.osgi.org/xmlns/blueprint/v1.0.0 http://www.osgi.org/xmlns/blueprint/v1.0.0/blueprint.xsd">
+
+    <bean id="sourcesToQueryPlugin"
+          class="org.codice.alliance.catalog.nsili.sourcestoquery.ui.SourcesToQueryPlugin"/>
+
+    <bean id="sourcesToQuery"
+          class="org.codice.alliance.catalog.nsili.sourcestoquery.ui.service.SourcesToQuery"
+          init-method="init"
+          destroy-method="destroy">
+        <property name="context" ref="blueprintBundleContext"/>
+        <property name="catalogFramework" ref="catalogFramework"/>
+        <property name="querySources" ref="querySources"/>
+    </bean>
+
+    <service interface="org.codice.ddf.admin.application.plugin.ApplicationPlugin"
+             ref="sourcesToQueryPlugin"/>
+
+    <reference id="querySources"
+               interface="org.codice.alliance.nsili.endpoint.QuerySources"
+               availability="mandatory"/>
+
+    <reference id="catalogFramework" interface="ddf.catalog.CatalogFramework"
+               availability="optional"/>
```

Comment:
I don't believe blueprint will ever give a null, because it always gives a proxy object. 
see `optional availability` in http://www.ibm.com/developerworks/library/os-osgiblueprint/